tests/api/v1/generatorTemplates/deleteWriters.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/api/v1/generatorTemplates/deleteWriters.js
+ */
+'use strict';
+
+const supertest = require('supertest');
+const api = supertest(require('../../../../index').app);
+const constants = require('../../../../api/v1/constants');
+const tu = require('../../../testUtils');
+const u = require('./utils');
+const expect = require('chai').expect;
+const GeneratorTemplate = tu.db.GeneratorTemplate;
+const User = tu.db.User;
+const writersPath = '/v1/generatorTemplates/{key}/writers';
+const writerPath = '/v1/generatorTemplates/{key}/writers/{userNameOrId}';
+
+describe('api: generatorTemplatess: permissions', () => {
+  let token;
+  let otherValidToken;
+  let generatorTemplate;
+  let user;
+  const generatorTemplateToCreate = u.getGeneratorTemplate();
+
+  beforeEach((done) => {
+    tu.createToken()
+    .then((returnedToken) => {
+      token = returnedToken;
+      done();
+    })
+    .catch((err) => done(err));
+  });
+
+  beforeEach((done) => {
+    GeneratorTemplate.create(generatorTemplateToCreate)
+    .then((gen) => {
+      generatorTemplate = gen;
+    }).then(() =>
```

Comment:
pls. see one of the updated tests from this weekend to see the formatting and structure of this boilerplate-ish beforeeach block